tests/store/model_registry/test_file_store.py

```diff
@@ -1,1364 +1,1341 @@
-            MlflowException,
-            match=(
-                r"While parsing a list in the query, "
-                r"expected string value, punctuation, or whitespace, "
-                r"but got different type in list"
-            ),
-        ) as exception_context:
-            search_versions("run_id IN (1,2,3)")
-        assert exception_context.value.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
 
-        assert set(search_versions(f"run_id LIKE '{run_id_2[:30]}%'")) == {2, 3}
+def test_rename_registered_model(store, registered_models_names, rm_data):
+    model_name = registered_models_names[random_int(0, len(registered_models_names) - 1)]
+    # Error cases
+    with pytest.raises(MlflowException, match="Registered model name cannot be empty."):
+        store.rename_registered_model(model_name, None)
+    # test that names of existing registered models are checked before renaming
 
-        assert set(search_versions(f"run_id ILIKE '{run_id_2[:30].upper()}%'")) == {2, 3}
+    for name in registered_models_names:
+        if name != model_name:
+            other_model_name = name
+            break
+    with pytest.raises(
+        MlflowException, match=rf"Registered Model \(name={other_model_name}\) already exists."
+    ):
+        store.rename_registered_model(model_name, other_model_name)
 
-        # search using the IN operator with empty lists should return exceptions
-        with pytest.raises(
-            MlflowException,
-            match=(
-                r"While parsing a list in the query, "
-                r"expected a non-empty list of string values, "
-                r"but got empty list"
-            ),
-        ) as exception_context:
-            search_versions("run_id IN ()")
-        assert exception_context.value.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
-
-        # search using an ill-formed IN operator correctly throws exception
-        with pytest.raises(
-            MlflowException, match=r"Invalid clause\(s\) in filter string"
-        ) as exception_context:
-            search_versions("run_id IN (")
-        assert exception_context.value.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
+    new_name = model_name + "!!!"
+    assert model_name != new_name
+    store.rename_registered_model(model_name, new_name)
+    assert store.get_registered_model(new_name).name == new_name
 
-        with pytest.raises(
-            MlflowException, match=r"Invalid clause\(s\) in filter string"
-        ) as exception_context:
-            search_versions("run_id IN")
-        assert exception_context.value.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
 
-        with pytest.raises(
-            MlflowException, match=r"Invalid clause\(s\) in filter string"
-        ) as exception_context:
-            search_versions("name LIKE")
-        assert exception_context.value.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
+def _extract_names(registered_models):
```

Comment:
this utility function can go above the first test

Reply:
moved to the begining, https://github.com/wkrt7/mlflow/commit/7460619d7d5762a055d0d9ebb309c873ca9bdb2a